### ISSUE TITLE
v3: Generated MarkDown tables are not understood by e.g. GitHub

### ISSUE THREAD

**Issue by andreasabel** [txt2tags/txt2tags#227]
Concerns target `md` in branch `v3`.

Consider this .txt file:
```txt2tags
Title1
Title2
Title3

|| Foo1 | Foo2 | Foo3 | Foo4 ||
| foo1 | foo2 | foo3 | foo4
```
In version 3.7, `txt2tags -t md` creates the following markdown file:
```md
Title1
Title2
Title3


| Foo1 |Foo2 |Foo3 |Foo4|
|---------------|
|foo1 |foo2 |foo3 |foo4|
```
This does not seem to be the correct table format, see https://www.markdownguide.org/extended-syntax/#tables
Such tables are not understood by e.g. GitHub:

| Foo1 |Foo2 |Foo3 |Foo4|
|---------------|
|foo1 |foo2 |foo3 |foo4|

The correct format would be (> 3 dashes ignored):
```md
| Foo1 |Foo2 |Foo3 |Foo4|
|---|---|---|------|
|foo1 |foo2 |foo3 |foo4|
```
This is understood:

| Foo1 |Foo2 |Foo3 |Foo4|
|---|---|---|------|
|foo1 |foo2 |foo3 |foo4|

**Comment by andreasabel** [comment]
The generic approach taken to tables does not seem to work for markdown.  There is a fixed separator between heading and rows https://github.com/txt2tags/txt2tags/blob/289ae320648b328dc9d04b5dec47282304c08874/txt2tags.py#L1268 when there should be a separator that is dependent on the number of columns.  Seems like the generic approach needs some refactoring that allows some of the markup strings to be dependent on the number of rows.

Likely, another backend could benefit from that as well, see this comment:
https://github.com/txt2tags/txt2tags/blob/289ae320648b328dc9d04b5dec47282304c08874/txt2tags.py#L715

**Comment by jendrikseipp** [comment]
I agree with your comments. It would be great if you could take a stab at this in a pull request.

**Comment by andreasabel** [comment]
I better leave this to Python programmers, I am a Haskeller.

**Comment by jendrikseipp** [comment]
Thanks for giving it a try!

**Comment by jendrikseipp** [comment]
This was fixed in #243.
